optimum_benchmark/scenarios/energy_star/scenario.py

```diff
@@ -182,79 +157,134 @@ def run(self, backend: Backend[BackendConfigT]) -> BenchmarkReport:
-        LOGGER.info("\t+ Running energy tracking")
+    # Loading tracking
+    def run_model_loading_tracking(self, backend: Backend[BackendConfigT]):
+        self.logger.info("\t+ Running model loading tracking")
 
-        with self.energy_tracker.track():
-            for inputs in tqdm(self.dataloader):
-                inputs = backend.prepare_inputs(inputs)
-                _ = backend.forward(inputs, self.config.forward_kwargs)
+        if self.config.energy:
+            energy_tracker = EnergyTracker(
+                backend=backend.config.name, device=backend.config.device, device_ids=backend.config.device_ids
+            )
 
-        self.report.prefill.energy = self.energy_tracker.get_energy()
-        self.report.prefill.efficiency = Efficiency.from_energy(
-            self.report.prefill.energy, self.text_generation_prefill_volume, unit=TEXT_GENERATION_EFFICIENCY_UNIT
-        )
-        self.energy_tracker.reset()
+        context_stack = ExitStack()
+        if self.config.energy:
+            context_stack.enter_context(energy_tracker.track())
+
+        with context_stack:
+            self.logger.info("\t+ Loading model for Inference")
+            backend.load()
 
-        with self.energy_tracker.track():
-            for inputs in tqdm(self.dataloader):
-                _ = backend.generate(inputs, self.config.generate_kwargs)
+        if self.config.energy:
+            self.report.load.energy = energy_tracker.get_energy()
 
-        self.report.decode.energy = self.energy_tracker.get_energy()
+    ## Energy tracking
+    def run_text_generation_energy_tracking(self, backend: Backend[BackendConfigT]):
+        self.logger.info(f"\t+ Running Text Generation energy tracking for {self.config.iterations} iterations")
+
+        prefill_measures = []
+
+        prefill_kwargs = {**self.config.generate_kwargs, **TEXT_GENERATION_PREFILL_OVERRIDES}
+
+        for k in range(self.config.iterations):
+            self.logger.info(f"\t+ Prefill iteration {k+1}/{self.config.iterations}")
+            with self.energy_tracker.track(file_prefix="prefill"):
+                prefill_volume = 0
+                for inputs in tqdm(self.dataloader):
+                    inputs = backend.prepare_inputs(inputs)
+                    _ = backend.prefill(inputs, prefill_kwargs)
+                    try:
+                        prefill_volume += inputs["input_ids"].size(dim=1) * self.config.input_shapes["batch_size"]
+                    except KeyError:
+                        prefill_volume += 1
+            prefill_measures.append(self.energy_tracker.get_energy())
+
+        prefill_energy = Energy.aggregate(prefill_measures)
+        self.report.prefill.energy = prefill_energy
+        self.report.prefill.efficiency = Efficiency.from_energy(
+            prefill_energy, prefill_volume, unit=TEXT_GENERATION_EFFICIENCY_UNIT
+        )
+        self.report.prefill.measures = prefill_measures
```

Comment:
Kinda ambiguous, and I'm not sure how the benchmark report behaves with a list of values ?
would it not be better to have an `Energy` class that can take multiple measures, or even an `EnergyStar` class that contains the whole thing